### DISCUSSION TITLE
Invalid URL

### DISCUSSION THREAD

**Discussion by smashrockwell** [mozilla/hubs#3568] [Hubs Cloud Questions] [answered]
I purchased Hubs Cloud Enterprise and went through the set up process with confirmations from the cloud formation system that everything is running fine but if I type in my URL I get Invalid URL. 

Why is my URL invalid and what should I do to fix this?

Thank you

**Comment by smashrockwell** [comment]
Whatever. Deleted that stack and just gonna redo it but this support system is lame, Mozilla. All this technology, set your support game up.

**Comment by emclaren** [comment]
Sorry to hear you're having trouble getting started with Hubs Cloud. We are a small team and are working on scaling up our ability to provide support. For now, we try to answer questions as best as we can. Ours is a new relatively product and delivering it via the Amazon Marketplace is new for us as well. As people encounter problems we help them solve them and improve our documentation. 

Its a bummer you are frustrated by the support systems available. We welcome feedback on our processes, but its hard to know what you mean when you say its "lame". 

There was a similar question asked today on support.mozilla.com - was this from you? If so, it would be helpful to know what lead you to post in both places. This seems like it would be frustrating to have to post in multiple channels and I would love to know if we can improve our documentation to reduce the likelihood of this happening.

**Reply by smashrockwell** [reply]
Thanks for the response. I was driven to post in multiple places because I have a problem I needed a remedy for today.  This information should be used to better your system but current I still do not have a remedy to my Route 53 domain name displaying "Invalid URL" on my homepage.

**Comment by robinkwilson** [comment]
**Which domain recipe are you following for your Hubs Cloud Deployment?** https://hubs.mozilla.com/docs/hubs-cloud-aws-domain-recipes.html

If you were following Recipe 3 (setting up SSL certificates on an existing domain), you have one more step to connect your domain to your Hubs Cloud after a successful stack deployment: https://hubs.mozilla.com/docs/hubs-cloud-aws-existing-domain.html#after-creating-the-stack-dns-setup

**Comment by smashrockwell** [comment]
I used Recipe 1. I purchased my domains from Route 53.

On Wed, Dec 16, 2020 at 5:18 PM Robin K Wilson <notifications@github.com>
wrote:

> *Which domain recipe are you following for your Hubs Cloud Deployment?*
> https://hubs.mozilla.com/docs/hubs-cloud-aws-domain-recipes.html
>
> If you were following Recipe 3 (setting up SSL certificates on an existing
> domain), you have one more step to connect your domain to your Hubs Cloud
> after a successful stack deployment:
> https://hubs.mozilla.com/docs/hubs-cloud-aws-existing-domain.html#after-creating-the-stack-dns-setup
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/hubs/discussions/3568#discussioncomment-217814>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABQXPXZWMBZ5SHWUVD2PQUTSVFL5VANCNFSM4U6N2YOA>
> .
>


-- 
Jonathan Owens
Hieroglyphics Imperium
1200 50th Ave,
Oakland Ca, 94601
Ph:510-710-0163

**Reply by smashrockwell** [reply]
last night I ran another instance of the Hubs app to a different URL. Once the stack was complete I visited the new URL and got the same results, invalid URL. All of my domain names are from Route 53. 

Please advise

**Reply by robinkwilson** [reply]
Awesome, Recipe 1. Did you follow the step by step video guide we have linked on the docs? It follows Recipe 1: https://hubs.mozilla.com/docs/hubs-cloud-aws-quick-start.html

Are you deploying with 2 domains? Or just 1?

Double check that your main domain name is correct with no caps.
Did any CNAME or A records get deleted on your domains after the stack spun up?
It's possible, you may need to delete the stack, then spin it back up - _happens occasionally when something doesn't get created quite right_

**Reply by smashrockwell** [reply]
Thanks for sending this video. It is more informative that the other "Setting up Hubs Cloud" video I followed. In this video the instructor explained the difference between .pen and .ppk for Key pairs. I am on windows but my keypair was .pen. That is the first error I noticed on my behalf. After I finish this video I will respin the stack with the proper keypair for windows and will report back on if that was the cause of my Invalid URL on my home page.

**Comment by smashrockwell** [comment]
Now when trying to spin a stack I get Rollback_in_Progress.

Three things I changed from watching the video are;

1. the file type of my key pair was changed to pkk.
2. I unlocked my URLs that were not unlocked for Transfer lock. (most were unlocked except the .link ones)
3. I tried to terminate 4 instances that were running. 

Now I get Rollback_in_Progress.

Just looked and the Rollback_is_Complete.

Please advise

**Reply by robinkwilson** [reply]
Can you scroll down to the FIRST red error message in your stack events? it should have some description of what error triggered the rollback to begin.

**Reply by robinkwilson** [reply]
Thanks for bearing with us :)

**Reply by smashrockwell** [reply]
Status reason is blank.  The rollback is not providing information for the Status Reason. 

I tried to re-spin this stack before I posted and that rolled back too. Tried again, to no avail. 


Thanks for the kind words

**Comment by smashrockwell** [comment]
The Rollback prior to this one did provided information. I had to scroll down a few pages. The most recent rollback did not provide information. Here is the previous rollback info (both attempts were this morning) 

The following resource(s) failed to create: [AppPostgrestDbSecret, InternalZoneSSLCertEast, AppDb, SESDomainRecords, StorageEFSAppMountTargetBPrivate, LinkRedirectorCloudfrontDistribution, StorageEFSAppMountTargetAPrivate]. Rollback requested by user.

**Reply by smashrockwell** [reply]
Here is another message from the previous rollback under LinkRedirectorCloudfrontDistribution;

One or more of the CNAMEs you provided are already associated with a different resource. (Service: AmazonCloudFront; Status Code: 409; Error Code: CNAMEAlreadyExists; Request ID: d2cd9b60-b698-480a-b177-246b913b9f05; Proxy: null)

**Reply by robinkwilson** [reply]
This one we can debug. Seems you need to delete any unused CNAMES.

**Reply by robinkwilson** [reply]
Do you have the stack running? Or did you delete the stack and now you're experiencing the Rollback issue after trying to deploy it again?

The CNAMES are likely leftover the cleanup and need to be manually deleted. Follow #4 here: https://hubs.mozilla.com/docs/hubs-cloud-aws-troubleshooting.html#potential-solutions

**Reply by robinkwilson** [reply]
But if you haven't deleted your stack yet, then delete it before deleting any CNAMEs.

**Comment by smashrockwell** [comment]
oh, I deleted the stack and now I'm experiencing the Rollback issue after trying to deploy it again.

When trying to follow #4 from the documentation, I get confused. IT says "Delete the CNAMEs from left from the last Hubs stack deployment -- Careful not to delete any CNAMEs that your domains use for another service"

How can I know which service a CName is used for?  How can I know which Cname was from the last stack deployed? There is no timestamp or grouping to show which Cname is which.

**Reply by smashrockwell** [reply]
What other services would my url need Cnames for? I am only using the urls for Hubs Cloud. 

 I've already deleted one Cname before I noticed there were two more for my main url. With no time stamp how do we tell Cnames apart?

Presumably, the Cnames issue is from me trying to spin a stack to deploy the app after already doing so and not deleting the initial Cnames right? If so,will solving this address the invalid url problem that I've had from my first build? I also bought 3 new urls and the same thing happen.

So could it be, me having the wrong Key pair file, plus not unlocking my urls caused the invalid url issue?

Then when trying to re-spin the stack for my second deployment of that app on the same url, without deleting the Cnames caused my rollback issue?

Then can I delete all the Cnames and restart the deployment process?

**Reply by smashrockwell** [reply]
I will delete all the Cnames. What else would they be there for?

 Maybe this can be worded differently, "Delete the CNAMEs from left from the last Hubs stack deployment -- Careful not to delete any CNAMEs that your domains use for another service". 

 I am going to deduce there is no other reason my URL needs any of these Cnames because I am only using the URL for HUBS service. I am deleted ALL the Cnames. Hope I do not create another problem. 

If what I presume is correct the documentation could say "Delete the CNAMEs left from the last Hubs stack deployment -- you can delete all of the Cnames if you purchased a Route 53 URL to be used strictly for this service.  Careful not to delete any CNAMEs that your domains use for another service"  

Again, I only presume this is the case. 

I will post my result for anyone troubleshooting this in the future.

**Comment by smashrockwell** [comment]
I delete all the Cnames and got rollbacks 3 more times.  New key pairs each time. No status reasons for the roll back.

Are there other records that need to be deleted from my internal domain? Since I purchase 2 urls for internal domains for different sites, I can compare between the internal domain records reserved for site one and two. For the first site internal domain I see 34 records. For the second internal domain I purchased after having problems I see 12 records. Now I can only presume there are more records that need to be deleted from the first domain. There must be duplicate records causing a rollback?

**Comment by smashrockwell** [comment]
So. If anyone else experiences this rollback problem after trying to re-spin a stack they should know more files than the Cnames need to be erased from the records. In my case it appears there were duplicate records in internal domain folder causing the system to rollback with no status reason. I only realized this because as stated I purchased more URLS and compared them after trying to deploy on new urls only once.

Now I am back at square one. The homepage is saying Invalid URL but the my-url.com/signin works and /admin work. 

Also, I looked in my main url records folder and only saw two records , one A record and one SOA record. This is after spawning 6 new stacks, that were rolled back, but at the same time 6 stacks from earlier today reappeared with timestamps and all, magically, saying complete. This was a result of me deleting the duplicated files and trying to spin a stack again. I confirmed this because each root stack I tried to spin had a different root stack name. The app that is active now is on a root stack with a different name than the most current one I tried to spin that got rolled back. So all of my information is restored. When I went in admin I already had avatars and scenes I populated earlier before deleting this build. But, the home URL is "invalid url".

Now I'm just about to learn how to write my own Cname record and will place it in the Hosted zones, my-url.com records folder because there is no Cname file in there being pointed to anything. I deleted one earlier when told to but when the stack "reappeared" it did not reproduce a Cname file for the main url folder. So I think once I learn to make my own Cname files I will be good.

**Comment by smashrockwell** [comment]
So, I made a Cname file and upon researching things I went into my outputs folder and clicked the link that my-url.com should be pointing to, and the address for my rood domain is an Invalid URl.  The AddressForRootDomain in my output folder, once clicked, displays "invalid url". 

Please advise.

**Comment by smashrockwell** [comment]
So. I stopped the roll back feature and the event list finally started giving a status reason;

The first failure was from ShortlinkZoneSSLCertEast. The status reason is "Failed to create resource. An error occurred (LimitExceededException) when calling the RequestCertificate operation: Error: you have reached your limit of 20 certificates in the last year." 

Everything after that failed with status reason being "Resource creation cancelled".



Last failure was; The following resource(s) failed to create: [KeymasterRole, EmptyBucketRole, AppDb, StackTopic, SESDomainRecords, ShortlinkZoneSSLCertEast, StorageEFSAppMountTargetBPrivate, InvokeStackTopicHandlerPermission, SpeelycaptorApplication, StorageEFSAppMountTargetAPrivate, SendEmailUser].

**Comment by smashrockwell** [comment]
I finally got this problem resolved. I will post the resolution later if the thread is still open. Thanks for the help. II know it gets busy. Hubs Cloud is a wonderful thing.

**Comment by smashrockwell** [comment, marked as answer]
I finally solved this problem. Troubleshooting this, I learned a lot about the app. One of my main issues was caused by, when rolling back a stack, files inside of the hosted zones that should have been erased were not. So when I tried to spin another stack, it gave me various issues and would rollback without status reason. This caused me to turn off the rollback feature so I could see what was going on. I had to manually remove the old records from the hosted zones and spin the stack again. So that solve the issue with records not being erased from the hosted zones causing rollbacks and failures. This started my second problem. 

When creating the second stack, I did not pick the same server region as I chose the first time. The first region I chose was Virginia in the second region was Oregon. This created additional problems. As you can see in the documentation, or not, I've seen it stated somewhere that you must be in the same region as your AWS verification. If not, You will have issues when trying to get out the sandbox. What I've learned troubleshooting with a few Amazon tech Is that sandboxes are regional also. In my instance, I was let out of the sandbox in the Virginia region, but was not let out of the sandbox in Oregon. So when I spun my second stack with Oregon as the region, after my AWS was verified in Virginia , I could not get any verification emails from my-url.com because AWS only took me out of the Virginia sandbox, not the sandbox for the Oregon region.  
 
Once I got over that everything was running fine. I could not pinpoint what caused the initial invalid URL. Seems there might not have been A record pointing back to WWW.my. URL.com.

**Reply by robinkwilson** [reply]
Ahhhhhhh yes that makes sense. Thanks for the detailed walkthrough above, a few notes:
- You don't need to recreated the SSH keypair each time
- Glad it's a new year so you didn't have to ask AWS to increase your certificate limit
- Great language change, I'll update the docs
- You need to be out of the sandbox in N. Virginia NOT the deploy region (oregon). All of the emails go through N. Virginia. (People run into this often. **Please request for SES increase in N. Virginia** :D )

www.myurl.com is a subdomain. so if you wanted www.myurl.com to go to hubs cloud your main domain would be www.myurl.com. If you put myurl.com as the main domain then went to www.myurl.com, you'll have an invalid url problem. 

If you setup myurl.com as the main domain and still wanted www.myurl.com to point to myurl.com, there's a record on the Hosted Zone that can do this.
